Mathlib/RingTheory/Polynomial/Chebyshev.lean

```diff
@@ -51,25 +51,46 @@ and do not have `map (Int.castRingHom R)` interfering all the time.
 * Prove minimax properties of Chebyshev polynomials.
 -/
 
-
-noncomputable section
-
 namespace Polynomial.Chebyshev
 set_option linter.uppercaseLean3 false -- `T` `U` `X`
 
 open Polynomial
 
-open Polynomial
+local macro "int_ring_nf" : tactic => `(tactic | (
+  try simp only [Int.negSucc_eq, Int.ofNat_eq_coe]
+  try push_cast
+  ring_nf))
 
 variable (R S : Type*) [CommRing R] [CommRing S]
 
-/-- `T n` is the `n`-th Chebyshev polynomial of the first kind -/
-noncomputable def T : ℕ → R[X]
+/-- `T n` is the `n`-th Chebyshev polynomial of the first kind. -/
+noncomputable def T : ℤ → R[X]
   | 0 => 1
   | 1 => X
-  | n + 2 => 2 * X * T (n + 1) - T n
+  | (n : ℕ) + 2 => 2 * X * T (n + 1) - T n
+  | -((n : ℕ) + 1) => 2 * X * T (-n) - T (-n + 1)
+  termination_by n => Int.natAbs n + Int.natAbs (n - 1)
 #align polynomial.chebyshev.T Polynomial.Chebyshev.T
 
+@[simp]
+theorem T_add_two : ∀ n, T R (n + 2) = 2 * X * T R (n + 1) - T R n
+  | (k : ℕ) => T.eq_3 R k
+  | -(k + 1 : ℕ) => by linear_combination (norm := int_ring_nf) T.eq_4 R k
+#align polynomial.chebyshev.T_add_two Polynomial.Chebyshev.T_add_two
+
+theorem T_add_one (n : ℤ) : T R (n + 1) = 2 * X * T R n - T R (n - 1) := by
+  linear_combination (norm := ring_nf) T_add_two R (n - 1)
```

Comment:
Isn't `ring_nf` the default?

Reply:
The default is `ring1`, but this doesn't normalize the indices of `T R`